### PR TITLE
Change how we run e2fsck to check ext filesystems

### PR DESCRIPTION
**Pull request by vpodzime**
The '-p' option means "Automatically repair ("preen") the file system.  This
option will cause e2fsck to automatically fix any filesystem problems that can
be safely fixed without human intervention." which is something we really
shouldn't do as part of reset()/populate(). We should use '-n' instead "Open the
filesystem read-only, and assume an answer of `no' to all questions." which
guaranties no changes to be made on the file system.

We might want to add the '-p' functionality back at some point, but it needs to
be explicitly triggered by the user code (e.g. Anaconda). I think we need to add
a 'clean' property and a 'repair' method to the formats.FS class so that users
can see where the problem is (if any) and explicitly trigger a safe fixup
attempt if they want to.

Related: rhbz#1170803

### CODE REVIEW

**Comment by AdamWill** [comment]
I don't think this really addresses what people care about in RHBZ 1170803. Avoiding modifying the filesystem is fine, but what people really *complain* about is that just running the check blocks the install process potentially for a very long time, especially when it runs on a disk you weren't planning to do anything with at all in the first place. We really need to stop running this check at all except when it's actually *needed* - before a resize operation.

**Comment by vpodzime** [comment]
@AdamWill I understand that. But there's no other way to find out whether the file system is clean or not. And blivet needs that information to provide reliable data/info about the file system. I think we need some kind of fast, cheap and noninvasive check + indicate to user (in anaconda) that the file system needs a proper check so that they can tell if we should run it or not.

**Comment by AdamWill** [comment]
"And blivet needs that information to provide reliable data/info about the file system."

Um. Does it? In the bug we came to the agreement (I thought) that the check was specifically added because in a new upstream release of e2fsprogs, they'd made it so `resize2fs -P` (which we use to estimate the minimum possible size of ext partitions) would not work unless the filesystem had been checked recently. Upstream agreed that that was a mistake and changed it.

The setup you describe would probably be neat if it were possible, but we've never had anything like it before and AFAIK this check was not added for any greater reason than '`resize2fs -P` needs it', and that is no longer true.

**Comment by vpodzime** [comment]
@AdamWill My understanding is that because it's not possible to resize an unclean file system blivet needs to check it to decide whether to mark it resizable or not. Which is described in the sources:

```
# We can't allow resize if the filesystem has errors.
try:
    self.do_check()
except FSError:
    self._resizable = False
    raise
```

File: blivet/formats/fs.py [hunk 1]
```diff
@@ -274,32 +274,25 @@ def update_size_info(self):
         self._min_instance_size = Size(0)
         self._resizable = self.__class__._resizable
 
-        # We can't allow resize if the filesystem has errors.
+        # try to gather current size info
```
Comment (mulkieran) [inline blivet/formats/fs.py]: Assuming this goes in, method comments in header should be updated to match code changes.

Comment (vpodzime) [inline blivet/formats/fs.py]: Good catch, thanks!

Comment (vpodzime) [inline blivet/formats/fs.py]: Fixed

**Comment by AdamWill** [comment]
We need to do something with this...anyone?

**Comment by vpodzime** [comment]
> We need to do something with this...anyone?

Actually, I just rebased the latest version of the changes to current *2.1-devel* and it looks like it fixes the issue(s) for me. Or am I missing something? I do admit I got lost a bit in all these things.

File: blivet/tasks/fsck.py [hunk 2]
```diff
@@ -114,7 +114,9 @@ class Ext2FSCK(FSCK):
                     128: "Shared library error."}
 
     ext = availability.E2FSCK_APP
-    options = ["-f", "-p", "-C", "0"]
+    # "Force checking even if the file system seems clean." (we might get false results otherwise)
+    # + "Open the filesystem read-only, and assume an answer of `no' to all questions."
+    options = ["-f", "-n"]
```
Comment (dwlehman) [inline blivet/tasks/fsck.py]: You're also turning off the progress output. Why do that?

File: blivet/formats/fs.py [hunk 3]
```diff
@@ -317,32 +315,25 @@ def update_size_info(self):
...
         try:
-            self.do_check()
-        except FSError:
+            if self._info.available:
+                self._current_info = self._info.do_task()
+        except FSError as e:
+            log.info("Failed to obtain info for device %s: %s", self.device, e)
+        try:
+            self._size = self._size_info.do_task()
+            self._min_instance_size = self._size
```
Comment (dwlehman) [inline blivet/formats/fs.py]: Seems a bit odd that setting `self._min_instance_size` is in the `try` block.

File: blivet/formats/fs.py [hunk 4]
```diff
@@ -317,32 +315,25 @@ def update_size_info(self):
...
+        except (FSError, NotImplementedError) as e:
+            log.warning("Failed to obtain current size for device %s: %s", self.device, e)
+
+        # We absolutely need a current size to enable resize. To shrink the
```
Comment (dwlehman) [inline blivet/formats/fs.py]: I'm not completely sure this is true, but it's probably a good indicator of having the tools and the state required to resize.

File: blivet/tasks/fsck.py (hunk 2 above)
Comment (vpodzime) [inline blivet/tasks/fsck.py]: What is the point of having it? In the logs...

File: blivet/formats/fs.py (hunk 3 above)
Comment (vpodzime) [inline blivet/formats/fs.py]: Well, it should only be (re)set here if we are updating ``self._size``

File: blivet/formats/fs.py (hunk 4 above)
Comment (vpodzime) [inline blivet/formats/fs.py]: Yeah, it's an old comment that I just moved.

**Comment by vpodzime** [comment]
> We just talked about this, and I'm not sure it makes sense to have XFS._resizable default to False. Are there any conditions (that this code handles) under which we would determine that we cannot resize an existing XFS filesystem?

I'm not aware of any. XFS does a FS check when being mounted. We are not doing anything like that here. And we probably never mark XFS as resizable (or at least I'm not aware of any code possibly doing so).

File: blivet/tasks/fsck.py (hunk 2 above)
Comment (dwlehman) [inline blivet/tasks/fsck.py]: Touche, sir.

File: blivet/formats/fs.py (hunk 3 above)
Comment (dwlehman) [inline blivet/formats/fs.py]: I would have put it in an `else` block to isolate/identify the code that might raise an exception, but it's not important.

Comment (vpodzime) [inline blivet/formats/fs.py]: Yeah, that's a good idea.

**Pull request merged**